plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/sessionconfig/CodeScanSessionConfig.kt

```diff
@@ -114,29 +116,42 @@ class CodeScanSessionConfig(
         return Payload(payloadContext, srcZip)
     }
 
-    fun getFilePayloadMetadata(file: VirtualFile): PayloadMetadata =
-        PayloadMetadata(
-            setOf(file.path),
-            file.length,
-            Files.lines(file.toNioPath()).count().toLong(),
-            file.programmingLanguage().toTelemetryType()
-        )
+    private fun getFilePayloadMetadata(file: VirtualFile): PayloadMetadata {
+        try {
+            return PayloadMetadata(
+                setOf(file.path),
+                file.length,
+                countLinesInVirtualFile(file).toLong(),
+                file.programmingLanguage().toTelemetryType()
+            )
+        } catch (e: Exception) {
+            cannotFindFile("${file.path} with file payload creation error: ${e.message}")
+        }
+    }
 
     /**
      * Timeout for creating the payload [createPayload]
      */
     fun createPayloadTimeoutInSeconds(): Long = CODE_SCAN_CREATE_PAYLOAD_TIMEOUT_IN_SECONDS
 
     private fun countLinesInVirtualFile(virtualFile: VirtualFile): Int {
-        val bufferedReader = virtualFile.inputStream.bufferedReader()
-        return bufferedReader.useLines { lines -> lines.count() }
+        try {
+            val bufferedReader = virtualFile.inputStream.bufferedReader()
+            return bufferedReader.useLines { lines -> lines.count() }
+        } catch (e: Exception) {
+            cannotFindFile("${virtualFile.path} with line count error: ${e.message}")
+        }
     }
 
     private fun zipFiles(files: List<Path>): File = createTemporaryZipFile {
         files.forEach { file ->
-            val relativePath = file.relativeTo(projectRoot.toNioPath())
-            LOG.debug { "Selected file for truncation: $file" }
-            it.putNextEntry(relativePath.toString(), file)
+            try {
+                val relativePath = file.relativeTo(projectRoot.toNioPath())
+                LOG.debug { "Selected file for truncation: $file" }
+                it.putNextEntry(relativePath.toString(), file)
+            } catch (e: Exception) {
+                cannotFindFile("${file.pathString} with zipping error: ${e.message}")
```

Comment:
For the definition of this function, would it be more readable to split the file name from the reason?

Reply:
Updated the function